# Unemployment Analysis with Python

## Oasis Infobyte Data Science Internship — Task 2

### Objective
The objective of this project is to perform exploratory data analysis (EDA) on unemployment data in India to identify regional and temporal trends, with a particular focus on understanding the impact of the COVID-19 pandemic on unemployment rates.

### Tech Stack
- Python
- Pandas
- NumPy
- Matplotlib
- Seaborn

In [15]:
# Import required libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## 1. Loading the Dataset

The unemployment dataset is loaded using Pandas. It contains unemployment-related data for different regions of India across multiple dates.

In [16]:
# Load the unemployment dataset
df = pd.read_csv("data/Unemployment in India.csv")

# Display the first 5 rows
df.head()

,Region,Date,Frequency,Estimated Unemployment Rate (%),Estimated Employed,Estimated Labour Participation Rate (%),Area
0,Andhra Pradesh,31-05-2019,Monthly,3.65,11999139.0,43.24,Rural
1,Andhra Pradesh,30-06-2019,Monthly,3.05,11755881.0,42.05,Rural
2,Andhra Pradesh,31-07-2019,Monthly,3.75,12086707.0,43.50,Rural
3,Andhra Pradesh,31-08-2019,Monthly,3.32,12285693.0,43.97,Rural
4,Andhra Pradesh,30-09-2019,Monthly,5.17,12256762.0,44.68,Rural


## 2. Dataset Inspection

Before performing data analysis, the dataset is inspected to understand its dimensions, column structure, data types, and overall statistical characteristics.

In [17]:
# Check the dimensions of the dataset
print("Dataset Shape:", df.shape)

# Display column names
print("\nColumn Names:")
print(df.columns.tolist())

Dataset Shape: (768, 7)

Column Names:
['Region', ' Date', ' Frequency', ' Estimated Unemployment Rate (%)', ' Estimated Employed', ' Estimated Labour Participation Rate (%)', 'Area']


In [18]:
# Display dataset information
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 7 columns):
 #   Column                                    Non-Null Count  Dtype  
---  ------                                    --------------  -----  
 0   Region                                    740 non-null    str    
 1    Date                                     740 non-null    str    
 2    Frequency                                740 non-null    str    
 3    Estimated Unemployment Rate (%)          740 non-null    float64
 4    Estimated Employed                       740 non-null    float64
 5    Estimated Labour Participation Rate (%)  740 non-null    float64
 6   Area                                      740 non-null    str    
dtypes: float64(3), str(4)
memory usage: 66.1 KB


In [19]:
# Display statistical summary of numerical columns
df.describe()

,Estimated Unemployment Rate (%),Estimated Employed,Estimated Labour Participation Rate (%)
count,740.000000,7.400000e+02,740.000000
mean,11.787946,7.204460e+06,42.630122
std,10.721298,8.087988e+06,8.111094
min,0.000000,4.942000e+04,13.330000
25%,4.657500,1.190404e+06,38.062500
50%,8.350000,4.744178e+06,41.160000
75%,15.887500,1.127549e+07,45.505000
max,76.740000,4.577751e+07,72.570000


## 3. Data Cleaning and Preprocessing

The dataset is cleaned before analysis by checking for missing and duplicate values, standardizing column names, and converting the date column into an appropriate datetime format.

In [20]:
# Check for missing values
df.isnull().sum()

Region                                      28
 Date                                       28
 Frequency                                  28
 Estimated Unemployment Rate (%)            28
 Estimated Employed                         28
 Estimated Labour Participation Rate (%)    28
Area                                        28
dtype: int64

In [21]:
# Check for duplicate rows
print("Number of duplicate rows:", df.duplicated().sum())

Number of duplicate rows: 27


In [22]:
# Remove leading and trailing spaces from column names
df.columns = df.columns.str.strip()

# Display cleaned column names
df.columns.tolist()

['Region',
 'Date',
 'Frequency',
 'Estimated Unemployment Rate (%)',
 'Estimated Employed',
 'Estimated Labour Participation Rate (%)',
 'Area']

In [25]:
print(df["Date"].head().tolist())

[' 31-05-2019', ' 30-06-2019', ' 31-07-2019', ' 31-08-2019', ' 30-09-2019']


In [26]:
# Convert the Date column to datetime format
df["Date"] = pd.to_datetime(df["Date"].str.strip(), format="%d-%m-%Y")

# Check the updated data type
df["Date"].dtype

dtype('<M8[us]')

In [27]:
# Remove rows where all values are missing
df.dropna(how="all", inplace=True)

# Reset the index after removing empty rows
df.reset_index(drop=True, inplace=True)

print("Dataset shape after removing empty rows:", df.shape)
print("\nMissing values after cleaning:")
print(df.isnull().sum())

Dataset shape after removing empty rows: (740, 7)

Missing values after cleaning:
Region                                     0
Date                                       0
Frequency                                  0
Estimated Unemployment Rate (%)            0
Estimated Employed                         0
Estimated Labour Participation Rate (%)    0
Area                                       0
dtype: int64


In [28]:
# Check for duplicate rows after removing empty rows
print("Duplicate rows after removing empty rows:", df.duplicated().sum())

Duplicate rows after removing empty rows: 0


### Data Cleaning Observations

- The original dataset contained 28 completely empty rows.
- These empty rows were removed, leaving **740 valid records and 7 columns**.
- After removing the empty rows, no missing values remained in the dataset.
- No duplicate records were found in the cleaned data.
- Column names were standardized by removing unnecessary leading and trailing spaces.
- The `Date` column was converted to datetime format for time-based analysis.

## 4. Feature Preparation for Analysis

Additional time-based features are extracted from the `Date` column to make it easier to analyze unemployment patterns across months and years.

In [29]:
# Extract year and month from the Date column
df["Year"] = df["Date"].dt.year
df["Month"] = df["Date"].dt.month_name()

# Display the first few rows
df[["Date", "Year", "Month"]].head()

,Date,Year,Month
0,2019-05-31,2019,May
1,2019-06-30,2019,June
2,2019-07-31,2019,July
3,2019-08-31,2019,August
4,2019-09-30,2019,September
